src/main/java/uiPages/ProductPage.java

```diff
@@ -0,0 +1,70 @@
+package uiPages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import helpers.WebElementHelper;
+
+import static helpers.WebElementHelper.areVisible;
+import static helpers.WebElementHelper.setFieldValue;
+
+public class ProductPage extends BasePage {
+
+    private static final String URL_PATH = "/product";
```

Comment:
done
before I just added the product path from the file to the URL_PATH, what are the benefits of using this method with the placeholder?

Reply:
That placeholder reflects the correct url. The product url is ~/product/product_name. So the  URL_PATH you first defined it didn't contain the /product_name part. Because we need to define the URL_PATH in a way that will match any product url we use the %s placeholder, which means that it's a string; we don't know its value, but it url must be /product/a_string